### PR TITLE
[Merged by Bors] - feat(Algebra/Central): definition of central simple algebra

### PR DESCRIPTION
**Pull request by jjaassoonn**
---

- [x] depends on: #14503

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/CentralSimple/Defs.lean [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+-/
+
+import Mathlib.RingTheory.SimpleRing.Defs
+import Mathlib.Algebra.Algebra.Subalgebra.Basic
+
+/-!
+# Characteristic predicate for central simple algebras
+
+In this file we define the predicate `IsCentralSimple K D` where `K` is a field
+and `D` is a (noncommutative) `K`-algebra.
```
Comment (kbuzzard) [inline Mathlib/Algebra/CentralSimple/Defs.lean:14]:
```suggestion
and `D` is a (not necessarily commutative) `K`-algebra.
```

File: Mathlib/Algebra/CentralSimple/Defs.lean [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Note that the predicate makes sense just for `K` a `CommRing` but it doesn't give the
+right definition; for a commutative ring base, one should use the theory of Azumaya algebras.
+This adds an extra layer of complication which we don't need. In fact ideals of `K`
+immediately give rise to nontrivial quotients of `D` so there are no central simple
+algebras in this case according to our definition.
```
Comment (kbuzzard) [inline Mathlib/Algebra/CentralSimple/Defs.lean:20]:
```suggestion
In fact ideals of `K` immediately give rise to nontrivial quotients of `D` so there are no central simple
algebras in this case according to our definition, if K is not a field. The theory of central simple algebras
really is a theory over fields.
```

File: Mathlib/Algebra/CentralSimple/Basic.lean [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+lemma tower {k K D : Type*} [Field k] [Field K] [Ring D]
+    [Algebra k K] [Algebra K D] [Algebra k D] [IsScalarTower k K D] [IsCentralSimple k D] :
+    IsCentralSimple K D where
+  is_central x := by
+    change x ∈ Subalgebra.center k D → _
+    rw [center_eq_bot k D, Algebra.mem_bot, Algebra.mem_bot]
+    simp only [Set.mem_range, forall_exists_index]
+    rintro x rfl
+    refine ⟨algebraMap k K x, by simp only [algebraMap_eq_smul_one, smul_assoc, one_smul]⟩
+  is_simple := IsCentralSimple.is_simple k
```
Comment (kbuzzard) [inline Mathlib/Algebra/CentralSimple/Basic.lean:51]: This lemma is a bit weird because necessarily k and K are both the centre of D...

Comment (jjaassoonn) [reply Mathlib/Algebra/CentralSimple/Basic.lean]: I have changed the lemma to say if `D/K/k` is a tower, then `D` central simple over `k` implies `K=k`

Comment (jjaassoonn) [reply Mathlib/Algebra/CentralSimple/Basic.lean]: It still feels a weird theorem though

**Comment by kbuzzard** [comment]
Ok lgtm!

maintainer merge

File: Mathlib/Algebra/CentralSimple/Defs.lean [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+universe u v
+
+/--
+For a field `K` and a `K`-algebra `D`, we say that `D` is a central algebra over `K` if the center
+of `D` is exactly `K` and that `D` is a simple ring.
+-/
+class Algebra.IsCentralSimple
+    (K : Type u) [Field K] (D : Type v) [Ring D] [Algebra K D] : Prop where
+  is_central : Subalgebra.center K D ≤ ⊥
+  [is_simple : IsSimpleRing D]
```
Comment (alreadydone) [inline Mathlib/Algebra/CentralSimple/Defs.lean:44]: How about introduce an Algebra.IsCentral class and combine it with IsSimpleRing? I think it makes little sense of bundling these two only, as you usually want to assume FiniteDimensional as well (to define the Brauer group etc.) 
Some results maybe holds more generally for artinian simple algebras, as those as semisimple.

File: Mathlib/Algebra/CentralSimple/Basic.lean [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+- `Algebra.IsCentralSimple.center_eq_bot`: the center of a central simple algebra over a field `K`
+  is equal to `K`.
+- `Algebra.IsCentralSimple.self`: a field is a central simple algebra over itself.
+- `Algebra.IsCentralSimple.baseField_essentially_unique`: Let `D/K/k` is a tower of scalars where
+  `K` and `k` are fields. If `D` is central simple over `k`, `K` is isomorphic to `k`.
+-/
+
+universe u v
+
+namespace Algebra.IsCentralSimple
```
Comment (alreadydone) [inline Mathlib/Algebra/CentralSimple/Basic.lean:26]: Results in this file still holds for IsCentral algebras, I think.

Comment (jjaassoonn) [reply Mathlib/Algebra/CentralSimple/Basic.lean:26]: the last one needs non-tiriviality, other than that everything generalizes

File: Mathlib/Algebra/CentralSimple/Defs.lean (hunk 4 above)
Comment (jjaassoonn) [reply Mathlib/Algebra/CentralSimple/Defs.lean]: I ended up just removing `Algebra.IsCentralSimple` as that instance does not imply `IsSimpleRing D` for not able to find synthesation orders

Comment (jjaassoonn) [reply Mathlib/Algebra/CentralSimple/Defs.lean]: I have added some docs

**Review by alreadydone — COMMENTED** [review]
LGTM otherwise

File: Mathlib/Algebra/Central/Defs.lean [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+we prove something is central simple, there we don't need to prove `⊥ ≤ center K D` though this
+direction is trivial.
+
+### Central Simple Algebras
+
+To define central simple algebras, we could do the following:
+```lean
+class Algebra.IsCentralSimple (K : Type u) [Field K] (D : Type v) [Ring D] [Algebra K D] where
+  [is_central : IsCentral K D]
+  [is_simple : IsSimpleRing D]
```
[the remark this answers is not in the document]
Comment (jjaassoonn) [reply Mathlib/Algebra/Central/Defs.lean:31]: Some results does not require finite dimensional assumption. I have added a bit more docs. Thanks

**Review by alreadydone — APPROVED** [review]
Thanks!
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**